### PR TITLE
[keymap] Added LipstickKeymap QML item. Fixes MER#1042

### PR DESCRIPTION
**Pull request by tanty**

### CODE REVIEW

**Comment by giucam** [comment]
I think i'd prefer if LipstickKeymap didn't subclass QWaylandKeymap but instead have it as a member. That way the new set* methods in QWaylandKeymaps aren't needed.

**Comment by tanty** [comment]
Any specific reason why not to extend the QWaylandKeymap class?

If I don't subclass, I would rather have 4 QString members that I can modify than a QWaylandKeymap member that I have to re-create every time it is modified.

If the problem is the delta against the upstream QWaylandKeymap I think I could file the patch also there.

**Comment by tanty** [comment]
Agreed to do the changes in Lipstick.

Closing ...

**Comment by tanty** [comment]
Argh ... the XF86Phone part is still valid ...

File: src/compositor/lipstickkeymap.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/***************************************************************************
+**
+** Copyright (C) 2015 Jolla Ltd.
+** Contact: Aaron Kennedy <aaron.kennedy@jollamobile.com>
```
Comment (mikkoharju) [inline src/compositor/lipstickkeymap.cpp]: s/aaronkennedy/tanty/g ?

Comment (tanty) [inline src/compositor/lipstickkeymap.cpp]: Having into account that I'm a Jolla subcontractor and this is under Jolla's copyright, I think it makes sense to keep the same contact person than in other files which are also under Jolla's copyright.

I can replace it with myself in any case, though.

File: src/compositor/lipstickcompositor.cpp [hunk 2]
```diff
@@ -622,20 +624,38 @@ void LipstickCompositor::setScreenOrientation(Qt::ScreenOrientation screenOrient
...
-    if (layout != m_keyboardLayout) {
-        m_keyboardLayout = layout;
-        defaultInputDevice()->setKeymap(QWaylandKeymap(m_keyboardLayout));
-        emit keyboardLayoutChanged();
+    if (!keymap) {
+        qWarning() << "Cannot set an empty keymap";
+        return;
+    }
+
+    if (!m_keymap || (*m_keymap != *keymap)) {
```
Comment (giucam) [inline src/compositor/lipstickcompositor.cpp]: The "(*m_keymap != *keymap)" is odd, imho. This method should set *that* keymap object, and return early only if we're setting the same object as the current one.

Comment (tanty) [inline src/compositor/lipstickcompositor.cpp]: Right. I will update.

**Comment by giucam** [comment]
LGTM

**Pull request merged**